sky/provision/nebius/utils.py

```
@@ -0,0 +1,281 @@
+                time.sleep(POLL_INTERVAL)
+
+    return wrapper
+
+
+def get_project_by_region(region: str) -> str:
+    service = ProjectServiceClient(sdk)
+    projects = service.list(ListProjectsRequest(parent_id=NB_TENANT_ID,)).wait()
+    for project in projects.items:
+        if region == 'eu-north1' and project.metadata.id[8:11] == 'e00':
+            return project.metadata.id
+        if region == 'eu-west1' and project.metadata.id[8:11] == 'e01':
+            return project.metadata.id
+    raise Exception(f'No project found for region "{region}".')
+
+
+def get_or_creat_gpu_cluster(name: str, project_id: str) -> str:
+    """Creates a GPU cluster."""
+    service = GpuClusterServiceClient(sdk)
+    try:
+        cluster = service.get_by_name(
+            GetByNameRequest(
+                parent_id=project_id,
+                name=name,
+            )).wait()
+        cluster_id = cluster.metadata.id
+    except RequestError:
+        cluster = service.create(
+            CreateGpuClusterRequest(
+                metadata=ResourceMetadata(
+                    parent_id=project_id,
+                    name=name,
+                ),
+                spec=GpuClusterSpec(infiniband_fabric='fabric-4'))).wait()
+        cluster_id = cluster.resource_id
+    return cluster_id
+
+
+def delete_cluster(name: str, project_id: str) -> None:
+    """Delete a GPU cluster."""
+    service = GpuClusterServiceClient(sdk)
+    try:
+        cluster = service.get_by_name(
+            GetByNameRequest(
+                parent_id=project_id,
+                name=name,
+            )).wait()
+        cluster_id = cluster.metadata.id
+        logger.debug(f'Found GPU Cluster : {cluster_id}.')
+        service.delete(DeleteGpuClusterRequest(id=cluster_id)).wait()
+        logger.debug(f'Deleted GPU Cluster : {cluster_id}.')
+    except RequestError as e:
+        logger.debug(f'GPU Cluster does not exist or can not deleted {e}.')
```

Should we raise an error instead of handling it transparently? If this happens, it could indicate resource leak and we would like to notify the user :))